scipy/optimize/tests/test_zeros.py

```diff
@@ -39,6 +42,39 @@ def test_brentq(self):
     def test_brenth(self):
         self.run_check(cc.brenth, 'brenth')
 
+    def test_sidi_k1(self):
```

Comment:
Could perhaps gain some concision by using a parametrized pytest for the value of `k` and just have one single `test_sidi_k` test.

Reply:
How is that specified for member functions? https://docs.pytest.org/en/2.9.0/parametrize.html only shows it for stand-alone functions.  [Separate question is then whether that functionality is supported in all the versions of pytest currently supported by SciPy...]